lib/IDE/Utils.cpp

```diff
@@ -1252,12 +1252,60 @@ Expr *swift::ide::getBase(ArrayRef<Expr *> ExprStack) {
   return Base;
 }
 
-bool swift::ide::isDynamicCall(Expr *Base, ValueDecl *D) {
-  auto TyD = D->getDeclContext()->getSelfNominalTypeDecl();
-  if (!TyD)
+bool swift::ide::isDeclOverridable(ValueDecl *D) {
+  auto *NTD = D->getDeclContext()->getSelfNominalTypeDecl();
+  if (!NTD)
     return false;
 
-  if (isa<StructDecl>(TyD) || isa<EnumDecl>(TyD) || D->isFinal())
+  if (!(isa<ClassDecl>(NTD) || isa<ProtocolDecl>(NTD))) {
+    // Only classes and protocols support overridding by subtypes.
+    return false;
+  }
+
+  if (D->isFinal() || NTD->isFinal() || NTD->isActor()) {
+    // Decls where either they themselves are final or their containing type
+    // is final cannot be overridden. Actors cannot be subclassed and thus the
+    // given decl also can't be overridden.
+    return false;
+  }
+
+  // While willSet/didSet etc can be "overridden" in a sense, they can't
+  // actually be called within the code. Treat getters/setters specially since
+  // they are more-directly called (even if not syntactically a call).
```

Comment:
I'll change this comment - as it turns out, `willSet` and `didSet` are *not* overridable. They are actually final and each will be called:
```
class A {
  var a: Int = 0 {
    didSet {
      print("A")
    }
  }
}

class B: A {
  override var a: Int {
    didSet {
      print("B")
    }
  }
}

B().a = 1
B().a = 2
```

Actually prints:
```
A
B
A
B
```

`Read` and `Modify` are overridable though, so I might change those to be the same as `Get` and `Set` - they're effectively equivalent but allow yielding the underlying storage instead.

Reply:
Oh, interesting. Good find 👍